spec/03-plugins/14-request-transformer/02-access_spec.lua

```diff
@@ -858,6 +1016,69 @@ for _, strategy in helpers.each_strategy() do
     end)
 
     describe("add", function()
+
+      describe("Host header", function()
+        it("preserve_host = true, no host change in plugin: preserved host", function()
+          local res = assert(proxy_client:send {
+            method  = "GET",
+            path    = "/get",
+            headers = {
+              ["Content-Type"] = "application/json",
+              host             = "preserve-host.test"
+            }
+          })
+          assert.response(res).has.status(200)
+          local json = assert.response(res).has.jsonbody()
+          local value = assert.has.header("host", json)
+          assert.equals("preserve-host.test", value)
+        end)
+
+        it("preserve_host = false, no host change in plugin: upstream host", function()
+          local res = assert(proxy_client:send {
+            method  = "GET",
+            path    = "/get",
+            headers = {
+              ["Content-Type"] = "application/json",
+              host             = "no-preserve-host.test"
+            }
+          })
+          assert.response(res).has.status(200)
+          local json = assert.response(res).has.jsonbody()
+          local value = assert.has.header("host", json)
+          assert.equals(upstream_host, value)
+        end)
+
+        it("preserve_host = true, add fails for Host: preserved host", function()
+          local res = assert(proxy_client:send {
+            method  = "GET",
+            path    = "/get",
+            headers = {
+              ["Content-Type"] = "application/json",
+              host             = "preserve-host-add-host.test"
+            }
+          })
+          assert.response(res).has.status(200)
+          local json = assert.response(res).has.jsonbody()
+          local value = assert.has.header("host", json)
+          assert.equals("preserve-host-add-host.test", value)
+        end)
+
+        it("preserve_host = false, add fails for Host: upstream host", function()
```

Comment:
Aren't those tests just reflecting that `add` for Host is simply a no-op as well? `add fails for Host` makes me think it would produce an error of some sort, but the code just skips it, like `remove` for example.